### PR TITLE
Marketplace - Update empty state for installed subscriptions table

### PR DESCRIPTION
**Pull request by kdevnel**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/pull/41066

The design calls for a specific component when the Installed subscriptions table is empty. By default, we use the default state offered by the `Table` component. This PR replaces that with the correct component and styling.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Either use an account with no installed subscriptions (uninstall any you already have) or change [this line](https://github.com/woocommerce/woocommerce/blob/8dedb28ffc676ea0dce3553f7f515d62d017f837/plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx#L103) to `rows={ [] } `
3. Run the build command (`pnpm run build --filter=woocommerce` from the `/woocommerce/plugins/woocommerce` folder)
4. Visit **My subscriptions** (wp-env URL: http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
5. See the new design matches the Figma - 2sXEw33oLyTZ7Mu3808sed-fi-2889%3A247946

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss [hunk 1]
```diff
@@ -81,7 +81,7 @@ a.woocommerce-marketplace__refresh-subscriptions {
 	border-radius: 2px;
 	border: none;
 	padding: 2px 8px;
-	margin-left: $grid-unit-15;
```
Comment (kdevnel) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss]: Picked up by my local linter - comes from the parent branch.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx [hunk 2]
```diff
@@ -71,6 +78,26 @@ export function InstalledSubscriptionsTable( props: {
...
+	if ( ! props.rows || props.rows.length === 0 ) {
+		const marketplaceBrowseURL = getNewPath( {}, MARKETPLACE_PATH, {} );
+		const noInstalledSubscriptionsHTML = createInterpolateElement(
+			__(
+				'No extensions or themes installed. <a>Browse the Marketplace</a>',
+				'woocommerce'
+			),
+			{
+				p: <p />,
+				a: <a href={ marketplaceBrowseURL }>your account</a>,
```
Comment (andfinally) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx]:
```suggestion
				a: <a href={ marketplaceBrowseURL } />,
```

I think the link content is accidentally copied from another link – fortunately the interpolation doesn't replace our proper text, but we should change this.

**Review by andfinally — CHANGES_REQUESTED** [review]
Looks great, thanks!

It slightly troubles me that the empty view appears when the component first mounts. Really we should show the loading state. 

https://github.com/woocommerce/woocommerce/assets/1647564/02435e5a-ef0b-4603-b51f-19e18a015b98

The Figma also doesn't seem to account for the empty state in the "Available to use" list. But we can worry about that later.

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx (hunk 2 above)
Comment (kdevnel) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx]: Nice spot! Resolved now. It annoys me that I need to add a linting ignore comment above links anytime we do this with `createInterpolateElement`. Either way, solved!

**Comment by kdevnel** [comment]
> It slightly troubles me that the empty view appears when the component first mounts. Really we should show the loading state.

Well spotted! I didn't spot this in my initial testing since it flashed so quickly. Resolved now by checking if the component `isLoading` and only displaying this if it isn't.

> The Figma also doesn't seem to account for the empty state in the "Available to use" list. But we can worry about that later.

Yeah, this was a tricky one. I didn't want to complicate this further, and I don't think we will see this in practice. I'd like to address it for completeness’s sake, but it seemed unnecessary for now; I believe everyone will always have WooPayments listed here since it's bundled with WooCommerce. Tested with a totally fresh Woo.com account and WooPayments still shows up.

Thanks for the feedback @andfinally!

**Review by andfinally — APPROVED** [review]
Thanks for the adjustments, works a treat! 💥

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx [hunk 3]
```diff
@@ -71,6 +78,27 @@ export function InstalledSubscriptionsTable( props: {
...
 
+	if ( ! props.isLoading && ( ! props.rows || props.rows.length === 0 ) ) {
+		const marketplaceBrowseURL = getNewPath( {}, MARKETPLACE_PATH, {} );
+		const noInstalledSubscriptionsHTML = createInterpolateElement(
+			__(
+				'No extensions or themes installed. <a>Browse the Marketplace</a>',
+				'woocommerce'
+			),
+			{
+				p: <p />,
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx]: I'm not sure where this is used in the original string. There isn't a `p` tag to replace. Am I correct in thinking that?

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss [hunk 4]
```diff
@@ -126,6 +126,14 @@ a.woocommerce-marketplace__refresh-subscriptions {
 	padding: 0 12px;
 }
 
+.woocommerce-marketplace__my-subscriptions .woocommerce-table.is-empty {
+	background: none;
+	border: 1px solid var(--lt-color-border-light);
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss]: On my display I can barely see this color. Designs say `$gray-100` but that gives me the wrong value. It should be `#f0f0f0` but SCSS turns it into `#101517`. That is odd. Can we reference `##f0f0f0` directly here?

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx [hunk 5]
```diff
@@ -71,6 +78,27 @@ export function InstalledSubscriptionsTable( props: {
...
+				// eslint-disable-next-line jsx-a11y/anchor-has-content
+				a: <a href={ marketplaceBrowseURL } />,
+			}
+		);
+
+		return (
+			<EmptyTable numberOfRows={ 4 }>
+				{ noInstalledSubscriptionsHTML }
+			</EmptyTable>
+		);
```
Comment (raicem) [inline plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx:98]: I haven't made your life easier when structuring the page. I think is a good compromise, even though it causes **Available to use** table to be visible at all times. I know you are aware of this per the discussion on this PR.

**Review by raicem — APPROVED** [review]
Works great, thank you @kdevnel!

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss (hunk 4 above)
Comment (mlaetitia) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss]: Yap - same here. I see we're using below `border: 1px solid var(--gutenberg-gray-100, #f0f0f0);`

**Review by mlaetitia — APPROVED** [review]
Besides the border color - it's definitely working as expected ! Approving

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx (hunk 3 above)
Comment (kdevnel) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/table.tsx]: Oops, that was an oversight. I missed removing it from an earlier prototype! Fixed in https://github.com/woocommerce/woocommerce/pull/41267/commits/7451bcefe84b44d5e05aeac7e7350cec8c53a508

File: plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss (hunk 4 above)
Comment (kdevnel) [reply plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/my-subscriptions.scss]: Thanks! The variables are changing again and caused a little confusion. Thanks for spotting this, though. I've amended it to align with Lae's suggestion. https://github.com/woocommerce/woocommerce/pull/41267/commits/7451bcefe84b44d5e05aeac7e7350cec8c53a508

**Pull request merged**